lang/py3/avro/tests/test_datafile.py

```diff
@@ -77,19 +77,22 @@
    {'value': {'car': {'value': 'head'}, 'cdr': {'value': None}}}),
 )
 
-CODECS_TO_VALIDATE = ('null', 'deflate')
+def get_codecs_to_validate():
+  codecs = ('null', 'deflate')
 
-try:
-  import snappy
-  CODECS_TO_VALIDATE += ('snappy',)
-except ImportError:
-  logging.warning('Snappy not present, will skip testing it.')
+  try:
+    import snappy
+    codecs += ('snappy',)
+  except ImportError:
+    logging.warning('Snappy not present, will skip testing it.')
 
-try:
-  import zstandard
-  CODECS_TO_VALIDATE += ('zstandard',)
-except ImportError:
-  logging.warning('Zstandard not present, will skip testing it.')
+  try:
+    import zstandard
+    codecs += ('zstandard',)
+  except ImportError:
+    logging.warning('Zstandard not present, will skip testing it.')
+
+  return codecs
```

Comment:
Could these imports be moved to the top and follow the pattern

```python
try:
    import snappy
    HAS_SNAPPY = True
except ImportError:
    HAS_SNAPPY = False

try:
    import zstandard
    HAS_ZSTANDARD = True
except ImportError:
    HAS_ZSTANDARD = False
```

and then use those booleans to set up `CODECS_TO_VALIDATE`?

Reply:
Sure! Update the PR.